pkg/cloudprovider/drift.go

```diff
@@ -119,3 +124,35 @@ func (c *CloudProvider) isImageVersionDrifted(
 	}
 	return "", nil
 }
+
+// isSubnetDrifted returns drift if the nic for this nodeclaim does not match the expected subnet
+func (c *CloudProvider) isSubnetDrifted(ctx context.Context, nodeClaim *corev1beta1.NodeClaim, nodeClass *v1alpha2.AKSNodeClass) (cloudprovider.DriftReason, error) {
+	expectedSubnet := lo.Ternary(nodeClass.Spec.VnetSubnetID == nil, options.FromContext(ctx).SubnetID, lo.FromPtr(nodeClass.Spec.VnetSubnetID))
+	nicName := instance.GenerateResourceName(nodeClaim.Name)
+
+	// TODO: Refactor all of AzConfig to be part of options
+	nic, err := instance.GetNic(ctx, c.instanceProvider.AZClient.NetworkInterfacesClient, options.FromContext(ctx).NodeResourceGroup, nicName)
+	if err != nil {
+		if sdkerrors.IsNotFoundErr(err) {
+			return "", nil
+		}
+		return "", err
+	}
+	nicSubnet := getFirstSubnetFromNic(nic)
```

Comment:
We should probably instead check that we are checking the primary ipConfiguration rather than just grabbing the first one. 

Although from a karpenter side of things, we should have consistency between all of the subnets set in all ipconfigurations so really any ip configuration works.